toolkit/smart-contracts/offchain/src/csl.rs

```diff
@@ -398,35 +398,27 @@ impl TransactionContext {
 		})
 	}
 
-	pub(crate) async fn for_payment_key_with_change_address<C: QueryLedgerState + QueryNetwork>(
-		payment_key: &CardanoPaymentSigningKey,
-		change_address: &Address,
-		client: &C,
-	) -> Result<TransactionContext, anyhow::Error> {
-		let payment_key = payment_key.clone().0;
-		let network = client.shelley_genesis_configuration().await?.network.to_csl();
-		let protocol_parameters = client.query_protocol_parameters().await?;
-		let payment_address = key_hash_address(&payment_key.to_public().hash(), network);
-		let payment_key_utxos = client.query_utxos(&[payment_address.to_bech32(None)?]).await?;
-		Ok(TransactionContext {
-			payment_key,
-			payment_key_utxos,
-			network,
-			protocol_parameters,
+	pub(crate) fn with_change_address(&self, change_address: &Address) -> Self {
```

Comment:
This is introduce to avoid communication with Ogmios when we only need to have different change address